src/Core/Bridge/Elasticsearch/DataProvider/CollectionDataProvider.php

```diff
@@ -45,14 +46,25 @@ final class CollectionDataProvider implements ContextAwareCollectionDataProvider
 
     /**
      * @param RequestBodySearchCollectionExtensionInterface[] $collectionExtensions
+     * @param ResourceMetadataFactoryInterface|ResourceMetadataCollectionFactoryInterface $resourceMetadataFactory
      */
-    public function __construct(Client $client, DocumentMetadataFactoryInterface $documentMetadataFactory, IdentifierExtractorInterface $identifierExtractor, DenormalizerInterface $denormalizer, Pagination $pagination, ResourceMetadataFactoryInterface $resourceMetadataFactory, iterable $collectionExtensions = [])
+    public function __construct(Client $client, DocumentMetadataFactoryInterface $documentMetadataFactory, IdentifierExtractorInterface $identifierExtractor = null, DenormalizerInterface $denormalizer, Pagination $pagination, $resourceMetadataFactory, iterable $collectionExtensions = [])
     {
         $this->client = $client;
         $this->documentMetadataFactory = $documentMetadataFactory;
+
+        if ($this->identifierExtractor) {
+            trigger_deprecation('api-platform', '2.7', sprintf('The service "%s" is deprecated and will be removed in 3.0.', IdentifierExtractorInterface::class));
```

Comment:
```suggestion
            trigger_deprecation('api-platform', '2.7', sprintf('Passing an instance of "%s" is deprecated and will not be supported in 3.0.', IdentifierExtractorInterface::class));
```

Reply:
@soyuka is going to delete this interface eventually

Reply:
Ok but the term "service" isn't appropriate. This class can be used without a DIC.